Storage/Storage/CoreData/CoreDataIterativeMigrator.swift

```diff
@@ -59,21 +68,46 @@ final class CoreDataIterativeMigrator {
         var debugMessages = [String]()
 
         do {
-            try steps.forEach { step in
-                let mappingModel = try self.mappingModel(from: step.sourceModel, to: step.targetModel)
-
-                // Migrate the model to the next step
+            // Perform all the migration steps and acquire the last _migrated_ destination URL.
+            let lastTempDestinationURL = try steps.reduce(sourceStoreURL) { currentSourceStoreURL, step in
+                // Log a message
                 let migrationAttemptMessage = makeMigrationAttemptLogMessage(step: step)
                 debugMessages.append(migrationAttemptMessage)
                 DDLogWarn(migrationAttemptMessage)
 
-                try migrateStore(at: sourceStore,
-                                 storeType: storeType,
-                                 fromModel: step.sourceModel,
-                                 toModel: step.targetModel,
-                                 with: mappingModel)
+                // Migrate to temporary URL
+                let tempDestinationURL = try migrate(step: step,
+                                                     sourceStoreURL: currentSourceStoreURL,
+                                                     storeType: storeType)
+
+                // Destroy the `currentSourceStoreURL` if it is a temporary URL since we
+                // will no longer need it.
+                if currentSourceStoreURL != sourceStoreURL {
+                    try persistentStoreCoordinator.destroyPersistentStore(at: currentSourceStoreURL,
+                                                                          ofType: storeType,
+                                                                          options: nil)
+                }
```

Comment:
Thanks for checking the file system! It's interesting it doesn't fully delete the files 🙄 but they are in temporary folder so that sounds fine. I can't think of any way to make the process better, the additional comments look great!